tests/grpc/config_test.go

```diff
@@ -0,0 +1,525 @@
+// Copyright 2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpc
+
+import (
+	"encoding/json"
+	"testing"
+
+	replay "github.com/pulumi/pulumi-terraform-bridge/testing/x"
+	"github.com/stretchr/testify/require"
+
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi-go-provider/infer"
+	config "github.com/pulumi/pulumi-go-provider/tests/grpc/config/provider"
+)
+
+// These inputs were created by running `pulumi up` with PULUMI_DEBUG_GRPC=logs.json in
+// ./config/consumer.
```

Comment:
just to check if I'm following correctly: You created the consumer just for the purpose of capturing these test cases and captured these logs before making the change to infer/configuration.go to verify that the change did not affect what was sent over the wire right?

Reply:
I created the consumer just for the purpose of capturing these test logs. I did one capture before and verified that existing behavior worked as expected. Changing out to `applyDefaults` changes the behavior. Specifically it applies defaults in more cases and it does so correctly on more types.